### PR TITLE
Media: add test for searchTerm states

### PR DESCRIPTION
**Pull request by diegovar**
Adds a test to verify that the `searchTerm` states don't pollute each other

### CODE REVIEW

**Review by winterDroid — APPROVED** [review]

File: assets/src/edit-story/app/media/local/reducer.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ const INITIAL_STATE = {
  * @param {Object} obj.payload The details of the action, specific to the action
  * @return {Object} The new state
  */
-function reducer(state = INITIAL_STATE, { type, payload }) {
+export function reducer(state = INITIAL_STATE, { type, payload }) {
```
Comment (swissspidy) [inline assets/src/edit-story/app/media/local/reducer.js:49]: Why this change? The `reducer` is already the default export. No need to have a named export as well.

The [`eslint-plugin-import/no-named-as-default`](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-named-as-default.md) rule now warns about this, as it might cause unintended effects.

File: assets/src/edit-story/app/media/media3p/reducer.js [hunk 2]
```diff
@@ -65,7 +65,7 @@ function reduceProviderStates(state, { type, payload }) {
  * @param {Object} obj.payload The details of the action, specific to the action
  * @return {Object} The new state
  */
-function reducer(state = INITIAL_STATE, { type, payload }) {
+export function reducer(state = INITIAL_STATE, { type, payload }) {
```
Comment (swissspidy) [inline assets/src/edit-story/app/media/media3p/reducer.js:68]: Same here, no need for the named export

File: assets/src/edit-story/app/media/local/reducer.js (hunk 1 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/local/reducer.js:49]: Good point, I have #3279 to address this.
